### DISCUSSION TITLE
Duplicate a file/folder

### DISCUSSION THREAD

**Discussion by khaitranhq** [nvim-tree/nvim-tree.lua#1695] [Q&A]
Like the title, can you create a function and mapping to a key to duplicate a selected file/folder?
For example, I want to duplicate a file that has the directory `/a/b/file.txt`, it should create a new file with the directory `/a/b/file_copy.txt`.

**Comment by gegoune** [comment]
Should be possible to write function using provided API. Please have a look if you can put something together, I can try at some point next week maybe.

**Reply by PSeitz** [reply]
A predefined function to map a key against would be nice

**Reply by alex-courtis** [reply]
I'm not sure we really need nvim-tree for that; `:w file_copy.txt` will do the job.

**Comment by alex-courtis** [comment]
@khaitranhq @PSeitz please test this feature: #2278

**Comment by khaitranhq** [comment]
Thank you @alex-courtis. I'm having a busy month so  I will try this feature later.
